File: tests/end-to-end/cloudwatch/golang/stack.go
package main

import (
	"fmt"

	cdk "github.com/aws/aws-cdk-go/awscdk/v2"
	cloudwatch "github.com/aws/aws-cdk-go/awscdk/v2/awscloudwatch"
	"github.com/aws/constructs-go/constructs/v10"
	"github.com/aws/jsii-runtime-go"
)

type CloudwatchStackProps struct {
	cdk.StackProps
	/// Environment used for this deployment.
	EnvironmentName *string
}

type CloudwatchStack struct {
	cdk.Stack
}

func NewCloudwatchStack(scope constructs.Construct, id string, props *CloudwatchStackProps) *CloudwatchStack {
	var sprops cdk.StackProps
	if props != nil {
		sprops = props.StackProps
	}
	stack := cdk.NewStack(scope, &id, &sprops)

	cloud_watch.NewCfnAlarm(
		stack,
		jsii.String("MyApi5xxErrorsAlarm"),
		&cloud_watch.CfnAlarmProps{
			AlarmDescription: jsii.String("Example alarm"),
			Namespace: jsii.String("AWS/ApiGateway"),
			Dimensions: &[]interface{}{
				&DimensionProperty{
					Name: jsii.String("ApiName"),
					Value: jsii.String("MyApi"),
				},
			},
			MetricName: jsii.String("5XXError"),
			ComparisonOperator: jsii.String("GreaterThanThreshold"),
			Statistic: jsii.String("Average"),
			Threshold: jsii.Number(0),
			Period: jsii.Number(900),
			EvaluationPeriods: jsii.Number(1),
			TreatMissingData: jsii.String("notBreaching"),
			AlarmActions: &[]*string{
				cdk.Fn_ImportValue(jsii.String(fmt.Sprintf("%vAlarmsTopicArn", props.EnvironmentName))),
			},
		},
	)

	return &CloudwatchStack{
		Stack: stack,
	}
}

File: tests/end-to-end/cloudwatch/python/stack.py
from aws_cdk import Stack
import aws_cdk as cdk
import aws_cdk.aws_cloudwatch as cloudwatch
from constructs import Construct

class CloudwatchStack(Stack):
  def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
    super().__init__(scope, construct_id, **kwargs)

    # Applying default props
    props = {
      'environmentName': kwargs.get('environmentName', 'dev'),
      'alarmThreshold': cdk.CfnParameter(self, 'alarmThreshold', 
        type = 'Number',
        default = str(kwargs.get('alarmThreshold', '0.005')),
        no_echo = True,
      ).value_as_number,
    }

    # Resources
    myApi5xxErrorsAlarm = cloudwatch.CfnAlarm(self, 'MyApi5xxErrorsAlarm',
          alarm_description = 'Example alarm',
          namespace = 'AWS/ApiGateway',
          dimensions = [
            {
              'name': 'ApiName',
              'value': 'MyApi',
            },
          ],
          metric_name = '5XXError',
          comparison_operator = 'GreaterThanThreshold',
          statistic = 'Average',
          threshold = props['alarmThreshold'],
          period = 900,
          evaluation_periods = 1,
          treat_missing_data = 'notBreaching',
          alarm_actions = [
            cdk.Fn.import_value(f"""{props['environmentName']}AlarmsTopicArn"""),
          ],
        )


